src/page-modules/contact/travel-guarantee/travelGuaranteeFormMachine.ts

```diff
@@ -0,0 +1,257 @@
+import { TransportModeType } from '@atb-as/config-specs';
+import { assign, fromPromise, setup } from 'xstate';
+import { Line } from '../server/journey-planner/validators';
+import { machineEvents, ReasonForTransportFailure } from '../machineEvents';
+import { InputErrorMessages, formInputValidator } from '../formInputValidator';
+
+export const fetchMachine = setup({
+  types: {
+    context: {} as {
+      isIntialAgreementChecked: boolean;
+      isBankAccountForeign: boolean;
+      stateSubmitted: 'car' | 'taxi' | 'other' | undefined;
+      transportMode: TransportModeType | undefined;
+      line: Line | undefined;
+      fromStop: Line['quays'][0] | undefined;
+      toStop: Line['quays'][0] | undefined;
+      date: string;
+      plannedDepartureTime: string;
+      kilometersDriven: string;
+      reasonForTransportFailure: ReasonForTransportFailure | undefined;
+      feedback: string;
+      firstName: string;
+      lastName: string;
+      email: string;
+      address: string;
+      postalCode: string;
+      phoneNumber: string;
+      city: string;
+      bankAccountNumber: string;
+      IBAN: string;
+      SWIFT: string;
+      errorMessages: InputErrorMessages;
+    },
+    events: machineEvents,
+  },
+  guards: {
+    validateInputs: ({ context }) => formInputValidator(context),
+  },
+  actions: {
+    setIsIntialAgreementChecked: assign({
+      isIntialAgreementChecked: ({ context }) =>
+        !context.isIntialAgreementChecked,
+    }),
+    setIsBankAccountForeign: assign({
+      isBankAccountForeign: ({ context }) => !context.isBankAccountForeign,
+    }),
+    setStateSubmitted: assign({
```

Comment:
I think you can work on the naming here. The undefined could probably be removed by proper typing.